src/styles/block/calendar.scss

```
@@ -0,0 +1,61 @@
+.calendar-page {
+  margin: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  min-height: $height-calendar;
+  height: 100vh;
+  width: 100%;
+
+  &__calendar {
+    width: $width-calendar;
+    height: $height-calendar;
+    border: 1px solid white;
+    flex-shrink: 0;
+  }
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+
+  &__day {
+    width: $size-box-day;
+    height: $size-box-day;
+  }
+
+  @for $i from 1 through 31 {
+    &__day:nth-child(#{$i})::before {
+      display: block;
+      content: '#{$i}';
+      text-align: center;
+      line-height: $size-box-day;
+      background-color: $c-box-day;
+      border: 1px solid black;
+      font-family: Arial, sans-serif;
+      font-size: 30px;
+    }
+
+    &__day:nth-child(#{$i})::before:hover {
+      background-color: #fff;
+    }
+  }
+
+  @for $j from 28 through 31 {
+    &--month-lengs-#{$j} {
+      :nth-child(n + #{$j+1})::before {
+        visibility: hidden;
+      }
+    }
+  }
+
+  @each $day, $shift in $week {
+    &--start-day-#{$day}::before {
+      display: block;
+      content: '';
+      height: $size-box-day;
+      width: #{$shift};
+    }
```

Why creating pseudo element here? (:

```suggestion
    &--start-day-#{$day} :first-child {
      margin-left: $shift;
    }
```

---

I'm not sure if it will work with your code, but trying hasn't hurt anybody yet!